### PR TITLE
Remove named tensors

### PR DESCRIPTION
**Pull request by georgeyiasemis**
Changes to use with no named tensors. Updates are only within **direct/data** and **direct/nn**  directories.   
Validation on calgary-campinas data from checkpoints match with stored metrics. Training also works fine.
@jonasteuwen can you please review?

### CODE REVIEW

File: direct/common/subsample.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # coding=utf-8
 # Copyright (c) DIRECT Contributors
 
+# pylint: disable = E1123
+
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply direct/common/subsample.py:5]: Where should it be? End of file?

File: direct/data/datasets.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 import pathlib
+import random
 from functools import lru_cache
-from typing import Any, Callable, Dict, List, Optional, Union
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
 import numpy as np
 from torch.utils.data import Dataset, IterableDataset
 
+from direct.data.generator import FakeMRIDataGenerator
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply direct/data/datasets.py:15]: You are right. Is FakeMRIData ok?

**Comment by jonasteuwen** [comment]
At the line where you want to disable it.

File: direct/functionals/challenges.py [hunk 3]
```diff
@@ -38,9 +38,9 @@ def fastmri_psnr(gt, pred):
     """Compute Peak Signal to Noise Ratio metric (PSNR) compatible with the FastMRI challenge."""
     gt = _to_numpy(gt)[:, 0, ...]
     pred = _to_numpy(pred)[:, 0, ...]
-    from skimage.measure import compare_psnr
+    from skimage.metrics import peak_signal_noise_ratio as psnr
 
-    out = compare_psnr(gt, pred, data_range=gt.max())
+    out = psnr(image_true=gt, image_test=pred, data_range=gt.max())
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply direct/functionals/challenges.py:43]: Yes. setup.py is ok. Maybe dockerfile needs to be updated

File: direct/data/datasets.py [hunk 4]
```diff
@@ -17,16 +22,257 @@
...
+            "encoding_size": encoding_size,
+            "reconstruction_size": reconstruction_size,
+        }
+        return metadata
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx: int) -> Dict[str, Any]:
+
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply direct/data/datasets.py:163]: Insert or remove whitespace?

File: direct/data/datasets.py [hunk 5]
```diff
@@ -17,16 +22,257 @@
...
+            filenames = [filenames[0] + f"{_:05}" for _ in range(1, self.sample_size + 1)]
+
+        current_slice_number = 0
+        for idx, filename in enumerate(filenames):
+
+            if len(filenames) < 5 or idx % (len(filenames) // 5) == 0 or len(filenames) == (idx + 1):
+                self.logger.info(f"Parsing: {(idx + 1) / len(filenames) * 100:.2f}%.")
+
+            num_slices = self.spatial_shape[0] if len(self.spatial_shape) == 3 else 1
+
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply direct/data/datasets.py:141]: Insert or remove whitespace?

File: direct/data/datasets.py [hunk 6]
```diff
@@ -17,16 +22,257 @@
...
+        ]
+        self.kspace_context = kspace_context if kspace_context else 0
+        self.ndim = 2 if self.kspace_context == 0 else 3
+
+        if self.kspace_context != 0:
+            raise NotImplementedError("3D reconstruction is not yet supported with FakeMRIBlobsDataset.")
+
+    def set_seed(self, seed):
+        np.random.seed(seed)
+        random.seed(seed)
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply direct/data/datasets.py:121]: Fixed.

File: projects/fastmri/base.yaml [hunk 7]
```diff
@@ -0,0 +1,98 @@
+# This model is a reproduction with some small changes of our winning algorithm in the Calgary-Campinas challenge
+# at MIDL2020. It has an improved sensitivity estimation and better metric logging capabilities based on DIRECT v0.2
+# features.
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply projects/fastmri/base.yaml]: Done

**Comment by jonasteuwen** [comment]
There is too much ;-)

**Review by jonasteuwen — APPROVED** [review]
LGTM

**Pull request merged**